cab-token-generator/javatests/com/google/auth/credentialaccessboundary/ClientSideCredentialAccessBoundaryFactoryTest.java

```diff
@@ -572,4 +589,334 @@ private static void triggerConcurrentRefresh(
+        cabBuilder
+            .addRule(CredentialAccessBoundary.AccessBoundaryRule.newBuilder()
+                         .setAvailableResource(
+                             "//storage.googleapis.com/projects/"
+                             + "_/buckets/example-bucket")
+                         .setAvailablePermissions(ImmutableList.of(
+                             "inRole:roles/storage.objectViewer"))
+                         .setAvailabilityCondition(
+                             CredentialAccessBoundary.AccessBoundaryRule
+                                 .AvailabilityCondition.newBuilder()
+                                 .setExpression(
+                                     "resource.name.startsWith('projects/_/"
+                                     + "buckets/example-bucket/objects/" +
+                                       "customer-a'") // No closing bracket
+                                 .build())
+                         .build())
+            .build();
+
+    assertThrows(CelValidationException.class,
+                 () -> { factory.generateToken(accessBoundary); });
+  }
+
+  @Test
+  public void generateToken_withSessionKeyNotBase64Encoded_failure() throws Exception {
+    MockStsTransportFactory transportFactory = new MockStsTransportFactory();
+    transportFactory.transport.setReturnAccessBoundarySessionKey(true);
+    transportFactory.transport.setAccessBoundarySessionKey("invalid_key");
+
+    ClientSideCredentialAccessBoundaryFactory.Builder builder =
+        ClientSideCredentialAccessBoundaryFactory.newBuilder();
+
+    ClientSideCredentialAccessBoundaryFactory factory =
+        builder
+            .setSourceCredential(getServiceAccountSourceCredentials(
+                mockTokenServerTransportFactory))
+            .setHttpTransportFactory(transportFactory)
+            .build();
+
+    CredentialAccessBoundary.Builder cabBuilder =
+        CredentialAccessBoundary.newBuilder();
+    CredentialAccessBoundary accessBoundary =
+        cabBuilder
+            .addRule(
+                CredentialAccessBoundary.AccessBoundaryRule.newBuilder()
+                    .setAvailableResource("//storage.googleapis.com/projects/"
+                                          + "_/buckets/example-bucket")
+                    .setAvailablePermissions(
+                        ImmutableList.of("inRole:roles/storage.objectViewer"))
+                    .setAvailabilityCondition(
+                        CredentialAccessBoundary.AccessBoundaryRule
+                            .AvailabilityCondition.newBuilder()
+                            .setExpression(
+                                "resource.name.startsWith('projects/_/"
+                                + "buckets/example-bucket/objects/customer-a')")
+                            .build())
+                    .build())
+            .build();
+
+    assertThrows(IllegalArgumentException.class,
+                 () -> { factory.generateToken(accessBoundary); });
```

Comment:
for this one, can you document where IllegalArgumentException may stem from. This is a runtimeexception and isn't marked as thrown by `generateToken()` so it may be confusing for future devs to figure out where it's coming from.

Reply:
This comes from the base64 decoding of the session key. Since the session key is from the server, I changed it to catch it and re-throw it as an IllegalStateExeption.